mavisp/modes.py

```diff
@@ -106,7 +117,44 @@ def parse_metadata(self, data_dir, system):
             out_metadata['gitbook_entry'] = metadata['gitbook_entry']
         else:
             out_metadata['gitbook_entry'] = ''
-
+    
+        try:
+            structure_source = metadata["structure_source"]
+            if structure_source not in self.structure_sources:
+                mavisp_criticals.append(
+                    MAVISpCriticalError(
+                        f"Invalid structure_source '{structure_source}' in metadata file. "
+                        f"Expected one of: {', '.join(self.structure_sources.keys())}"
+                    )
+                )
+            out_metadata["structure_source"] = structure_source
+            out_metadata["structure_description"] = self.structure_sources.get(structure_source, None)
+    
+            linker_design = metadata["linker_design"]
+            if not isinstance(linker_design, bool):
+                mavisp_criticals.append(
+                    MAVISpCriticalError(f"Invalid value for linker_design: {linker_design}. Must be True or False.")
+                )
+            out_metadata["linker_design"] = linker_design
+    
+            try:
+                pdb_id = metadata["pdb_id"]
+                if structure_source in ["PDB", "Mod"] and not pdb_id:
+                    mavisp_criticals.append(
+                        MAVISpCriticalError(f"structure_source '{structure_source}' requires a non-empty 'pdb_id' field.")
+                    )
+                out_metadata["pdb_id"] = pdb_id if structure_source in ["PDB", "Mod"] else None
+            except KeyError:
+                if structure_source in ["PDB", "Mod"]:
+                    mavisp_criticals.append(
+                        MAVISpCriticalError(f"structure_source '{structure_source}' requires a 'pdb_id' field, but it was not found.")
+                    )
+                out_metadata["pdb_id"] = None
```

Comment:
we can restructure this a little:

  * first we check if the structure_source is PDB or Mod. If it is not, then we can just set pdb_id to None
  * then if we need we check if the pdb_id key exists. If it doesn't, we can handle the case
Always keep the `except` near their respective `try` so that we can catch specific conditions/cases

so like:

```
    if structure_source in ["PDB", "Mod"]:
        try:
            pdb_id = metadata["pdb_id"]
        except:
            ...
            pdb_id = None

        if len(str(pdb_id)) == 0:
            # handle case appropriately

        out_metadata["pdb_id"]

    else:
    
        out_metadata["pdb_id"] = None
```